### PR TITLE
Implement MusicRoom methods

### PR DESCRIPTION
**Pull request by toadster172**
Implements

- `MusicRoom::CheckInputEnable`
- `MusicRoom::ProcessInput`
- `MusicRoom::RegisterChain`
- `MusicRoom::MusicRoom`
- `MusicRoom::OnUpdate`
- `MusicRoom::OnDraw`
- `MusicRoom::AddedCallback`
- `MusicRoom::DeletedCallback`

Everything matches 100% except for RegisterChain and the constructor, which both have stack differences due to some missing variables, likely in another class's constructor.

### CODE REVIEW

File: objdiff.json [hunk 1]
```diff
@@ -1,44 +1,37 @@
 {
-    "custom_make": "python",
```
Comment (roblabla) [inline objdiff.json:2]: This updated objdiff to the latest version, which is incompatible with the objdiff used in CI ^^

File: scripts/configure.py [hunk 2]
```diff
@@ -117,6 +117,7 @@ def configure(build_type):
             "main",
             "GameWindow",
             "MidiOutput",
+            "MusicRoom",
             "Supervisor",
             # MusicRoom
```
Comment (roblabla) [inline scripts/configure.py:122]: Remove this comment? Also, I think musicroom goes after Supervisor

File: objdiff.json [hunk 3]
```diff
@@ -26,14 +26,6 @@
 
             "reverse_fn_order": false
         },
-        {
```
Comment (roblabla) [inline objdiff.json:29]: Why remove AnmVm.obj? This breaks the comparison step.

Comment (toadster172) [reply objdiff.json:29]: It was unintentional. I had restored the version in my local repository's upstream/master branch, but forgot to sync with the remote beforehand and ended up restoring an older version of the file. Sorry about that.

**Pull request merged**